libvast/vast/min_max_synopsis.hpp

```diff
@@ -40,7 +41,40 @@ class min_max_synopsis : public synopsis {
       max_ = *y;
   }
 
-  bool lookup(relational_operator op, data_view rhs) const override {
+  std::optional<bool> lookup(relational_operator op,
+                             data_view rhs) const override {
+    if (op == in) {
+      auto s = caf::get_if<view<set>>(&rhs);
+      if (s == nullptr)
+        return std::nullopt;
+      for (auto element : **s)
+        if (lookup_impl(op, element))
+          return true;
+      return false;
+    } else if (op >= equal && op <= greater_equal) {
```

Comment:
This relies on the enum order of `relational_operator`. We shouldn't assume an order over these types and only compare them with `==` or `!=`. If you need to perform a bulk comparison, you can always add a little helper function.

Reply:
Counterproposal: We should factor `relational_operator`. It itself should only contain the equality and ordering types. `match` and `not_match` should be in its own category, and the rest are `membership_operators`.

Reply:
We might look into hoisting the operators into the type domain at a later point in time. This will be resolved by rewriting as a switch..case expression.